presto-main/src/main/java/com/facebook/presto/sql/gen/SwitchCodeGenerator.java

```diff
@@ -97,49 +81,84 @@ else if (<value> == <non-constant2>) {
         // determine the type of the value and result
         Class<?> valueType = value.getType().getJavaType();
 
+        // We generate SearchedCase as CASE TRUE WHEN p1 THEN v1 WHEN p2 THEN p2...
+        boolean searchedCase = (value instanceof ConstantExpression && ((ConstantExpression) value).getType() == BOOLEAN &&
```

Comment:
Can we move the declaration of `value` here or move this up? It took me a while to find where value is defined 😂

Reply:
Done